src/gwt/src/org/rstudio/studio/client/workbench/views/source/editors/text/TextEditingTarget.java

```diff
@@ -3283,6 +3283,42 @@ private boolean isSelectionValueEmpty(String selection)
       return selection == null || selection.trim().length() == 0;
    }
 
+   @Handler
```

Comment:
But it doesn't seem that these commands are useful in the console, maybe with exception of `onBlockIndent/onBlockOutdent`?

Reply:
Still, it's preferable to have the commands implemented there so that they work with all Ace instances (not just those that happen to live in the Source pane)

Reply:
Should I also move `onReindent`, `onSplitIntoLines` etc while working on this PR?

Reply:
I think that's worth doing.